.github/workflows/main.yml

```diff
@@ -347,6 +347,7 @@ jobs:
       PLATFORM: linux64
       TEST: ${{matrix.target}}
       HXCPP_COMPILE_CACHE: ~/hxcache
+      HAXE_STD_PATH: /usr/local/share/haxe/std
```

Comment:
Why is this necessary?

Reply:
On unix, when this is not set we end up with 3 std path https://github.com/HaxeFoundation/haxe/blob/development/src/compiler/compiler.ml#L185

When our CI runs misc tests on mac, the second will be used for eval error reporting; on linux, the third one (and on windows, the first and only one).

Setting `HAXE_STD_PATH` allowed me to have a predictable std path to provide to misc tests and use to check error reporting [on std types](https://github.com/HaxeFoundation/haxe/blob/ca40974ad12828d2d1caccee9cd2c79e74dc156e/tests/misc/projects/Issue10844/user-defined-meta-json-indent-fail.hxml.stderr)

... which reminds me I couldn't add expected test output for those back in the day; will add that now.
Do you want me to extract this std path exposed to misc test business in a separate PR?